### DISCUSSION TITLE
Full Emulation of LMN 3

### DISCUSSION THREAD

**Discussion by atardigradestory** [FundamentalFrequency/.github#11] [Q&A] [answered]
Is it possible to both generate the MIDI signals with the emulator and send those signals to the LMN 3 DAW running on a PC? Also, I assume the "executable" releases in the DAW repository need to be run on a linux OS? If I am not mistaken and this is possible I am working on getting it running on a virtual machine (as my home computer runs windows). If I can get it working I'd be happy to put out a guide so that people who don't have a raspberry pi yet can still test out the LMN 3 software/firmware/UI etc.

**Comment by stonepreston** [comment, marked as answer]
this is indeed possible, you can see this in action in [this video](https://www.youtube.com/watch?v=HBd83-JuzLo). Using the emulator relies on using MIDI loopback that works out of the box on linux (you can send and receive from the same default Linux midi device.) 

yes you are correct the releases are for Linux only. you might experience some issues getting this to work in a VM but it might also go quite smoothly. I have not tried this before. Definitely document things as you go so you can post your experience. 

Since you are running windows, you should also be able to build the DAW and emulator from source provided you install CMake and a C++ compiler. Not sure how all that works on windows but it shouldnt be too hard. Although I will say doing things like this on WIndows usually requires jumping through a few hoops. 

In the long run, i would recommend just getting an external HD or something to install Linux on for development purposes. It will be much easier to develop on Linux running on your own metal than in a VM.

**Reply by atardigradestory** [reply]
This is a good suggestion. SSDs are so cheap nowadays, even *now,* that there's no reason not to. I didn't catch that video-- I'll go check it out. Will keep updates as I see success/failure on either of these fronts (compiling on windows or VM). Very eager to get my hands into the code. Apologies for going off topic, but I see so far there is just a sampler, drum sampler, and a 4Osc synth engine, is that right? I assume it is possible to develop more complex plugins. I'd love to work on expanding the synth engine options a bit-- a harmonics based organ-like engine, maybe an envelope-noise based drum engine, etc.

**Reply by stonepreston** [reply]
@atardigradestory yes additional plugins are possible. There is some support for VSTs in the LMN-3-DAW (VERY EXPERIMENTAL) but tbh it would be better to write "native" tracktion engine plugins in these early stages as they just easier to deal with. [Here](https://github.com/Tracktion/tracktion_engine/blob/develop/tutorials/04%20-%20DistortionEffectDemo.md) is a tracktion engine tutorial on creating a native plugin effect.

Fair warning, the tracktion engine is complicated and not super well documented (nothing on the level of the JUCE tutorials or anything). But with some perseverance you can wrap your head around things. Don't get discouraged, its tough at first. 

I suggest you start with JUCE tutorials first, then move to tracktion engine.

**Comment by atardigradestory** [comment]
As soon as I had the VM working the emulator and DAW were able to run without any hitches so far. I've run into some hiccups with GPU forwarding through the VM but that has nothing to do with the LMN software. I'd post a tutorial but seeing as all I've had to do was make a VM and there's plenty of info on that out there I'll spare it unless someone sees this and asks for help.

**Reply by stonepreston** [reply]
awesome! glad it wasnt too bad
